supporting/concentration_layer.py

```diff
@@ -231,6 +241,77 @@ def process_emissions(self, emis, isrm_obj, verbose, output_dir, output_emis_fla
             
         return tmp_dct['PM25'], tmp_dct['NH3'], tmp_dct['VOC'], tmp_dct['NOX'], tmp_dct['SOX']
     
+    def visualize_emissions(self, pollutant_name=''):
+
+        ''' Create a 5-panel plot of total emissions for each pollutant and save as a PNG file '''
+        
+        if self.verbose:
+            logging.info('- Drawing map of total emissions by pollutant.')
+
+        # Read in CA boundary
+        ca_shp = gpd.read_feather(self.shp_path)
+        ca_prj = ca_shp.to_crs(self.crs)
+        
+        # Reproject output_region
+        output_region = self.output_region.to_crs(self.crs)
+
+        pollutants = {
+            'Primary PM2.5': self.PM25e,
+            'NH3': self.NH3e,
+            'NOx': self.NOXe,
+            'SOx': self.SOXe,
+            'VOC': self.VOCe
+        }
+
+        fig, axes = plt.subplots(nrows=1, ncols=5, figsize=(22,6))
+        
+        for ax, (pol, data) in zip(axes, pollutants.items()):
+            
+            # Clip to output region if provided
+            if output_region is not None:
+                data = gpd.clip(data, output_region)
+            
+            sns.set_theme(context="notebook", style="whitegrid", font_scale=1.25)
+
+            data.plot(column='EMISSIONS_UG/S',
+                        legend_kwds={'label': "Emissions (ug/s)"},
+                        legend=True, 
+                        cmap='mako_r',
+                        edgecolor='none',
+                        antialiased=False,
+                        ax=ax)
+
+
+            output_region.boundary.plot(ax=ax, edgecolor='black', facecolor = 'none')  
+            
+            # Add north arrow
+            ax.annotate('', xy=(0.94, 0.95), xytext=(0.94, 0.92), arrowprops=dict(facecolor='black', shrink=0.4),
```

Comment:
Is this always north? I think the answer is probably yes because we're forcing it to the ISRM grid, right? Maybe just check the code pipeline to 100% confirm that this is always true. Also, does the ISRM's CRS actually orient with true north? Projections are hard!

Reply:
I've emailed Chirag and I've written code that calculates the angle for a north arrow given a custom CRS. I know that north isn't always north when using a custom CRS, but I just followed up asking if it's also the case for ISRM's CRS. If it isn't, should I have the code somewhere regardless? There's two methods: one for calculating the angle, and one for creating the rotated arrow!

Reply:
I think it's probably best practice to always calculate the angle and make sure it is true, true north. The two methods you described seem like a great approach!